###  Данные

In [2]:
import pandas as pd
import numpy as np
from statsmodels.stats.proportion import proportions_ztest

# Общие результаты
data_total = pd.DataFrame({
    'surface': ['Clay', 'Nonclay'],
    'wins': [1660, 3658],
    'total': [2523, 6373]
})

# Подача
data_serve = pd.DataFrame({
    'surface': ['Clay', 'Nonclay'],
    'wins': [1090, 2747],
    'total': [1261, 3193]
})

# Приём
data_return = pd.DataFrame({
    'surface': ['Clay', 'Nonclay'],
    'wins': [570, 911],
    'total': [1262, 3180]
})

### Функция A/B теста

In [4]:
def ab_test(df, title):
    print(f"\n===== {title} =====")
    
    # доли
    df['win_rate'] = df['wins'] / df['total']
    print(df[['surface', 'win_rate']])
    
    # z-test
    count = df['wins'].values
    nobs = df['total'].values
    
    stat, pval = proportions_ztest(count, nobs)
    
    print(f"\nZ-stat: {stat:.4f}")
    print(f"P-value: {pval:.6f}")
    
    if pval < 0.05:
        print("Различие значимо")
    else:
        print("Различие не значимо")

### Запуск тестов

In [5]:
ab_test(data_total, "Общие результаты")
ab_test(data_serve, "Геймы на подаче")
ab_test(data_return, "Геймы на приёме")


===== Общие результаты =====
   surface  win_rate
0     Clay  0.657947
1  Nonclay  0.573984

Z-stat: 7.2798
P-value: 0.000000
Различие значимо

===== Геймы на подаче =====
   surface  win_rate
0     Clay  0.864393
1  Nonclay  0.860319

Z-stat: 0.3546
P-value: 0.722911
Различие не значимо

===== Геймы на приёме =====
   surface  win_rate
0     Clay  0.451664
1  Nonclay  0.286478

Z-stat: 10.5320
P-value: 0.000000
Различие значимо


### Выводы

1. Общие результаты

Доля выигранных геймов на грунте (65.8%) статистически значимо выше, чем на негрунтовых покрытиях (57.4%)
(p-value ≈ 0)

2. Геймы на подаче

Различие между грунтом (86.4%) и негрунтом (86.0%) оказалось статистически незначимым (p-value = 0.72)

3. Геймы на приёме

На грунте Надаль выигрывает 45.2% геймов на приёме, тогда как на других покрытиях — только 28.6%.
Различие статистически значимо (p-value ≈ 0)